### DISCUSSION TITLE
How to disable Content-MD5 header with S3 PutObject request

### DISCUSSION THREAD

**Discussion by vbrajyaguru** [aws/aws-sdk-cpp#2926] [Q&A] [answered]
Hello,

I'm looking for a way to disable content verification on uploads. I'm using `PutObjectRequest` and set `ChecksumAlgorithm::NOT_SET` but the request still generated content-md5 header.

Aws SDK 1.11.293
SLES 15 SP4

**Comment by jmklix** [comment]
Currently there is no way to send a request without a checksum. If you want I can change this to a feature request? If we do make this a feature request I'm not sure when it would be completed.

**Reply by vbrajyaguru** [reply]
Yes, please consider this as feature request. We are using S3 client and using `PutObjectRequest` to create the PUT request. Our application has its own checksumming method to verify data integrity of the object read from the S3. We don't need data integrity protection while uploading the object to S3 and so we need some way to tell S3 client to not perform checksumming on the content.

Let me know and I can provide code samples if needed.

**Reply by jmklix** [reply]
Created feature request here: https://github.com/aws/aws-sdk-cpp/issues/2933

**Reply by DmitriyMusatkin** [reply, marked as answer]
are you using one of the algorithms currently provided by SDK (md5, crc32, crc32c, sha1 or sha256) for your own integrity verification solution? You could consider providing the value to SDK (that will avoid perfoming calculation on SDK side) as a work around for now.
